packages/twenty-server/src/engine/api/graphql/workspace-schema-builder/utils/get-available-aggregations-from-object-fields.util.ts

```diff
@@ -0,0 +1,94 @@
+import { GraphQLISODateTime } from '@nestjs/graphql';
+
+import { GraphQLFloat, GraphQLString } from 'graphql';
+
+import { FieldMetadataInterface } from 'src/engine/metadata-modules/field-metadata/interfaces/field-metadata.interface';
+
+import { FieldMetadataType } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { capitalize } from 'src/utils/capitalize';
+
+enum AGGREGATION_OPERATIONS {
+  min = 'MIN',
+  max = 'MAX',
+  avg = 'AVG',
+  sum = 'SUM',
+}
+
+type AggregationValue = {
+  type: typeof GraphQLString;
+  description: string;
+  fromField: string;
+  aggregationOperation: AGGREGATION_OPERATIONS;
+};
+
+export type AggregationField = {
+  [key: string]: AggregationValue;
+};
+
+export const getAvailableAggregationsFromObjectFields = (
+  fields: FieldMetadataInterface[],
+): AggregationField[] => {
+  return fields.reduce<Array<Record<string, any>>>((acc, field) => {
+    if (field.type === FieldMetadataType.DATE_TIME) {
+      return [
+        ...acc,
+        {
+          [`min${capitalize(field.name)}`]: {
+            type: GraphQLISODateTime,
+            description: `Oldest date contained in the field ${field.name}`,
+            fromField: field.name,
+            aggregationOperation: AGGREGATION_OPERATIONS.min,
+          },
+        },
+        {
+          [`max${capitalize(field.name)}`]: {
+            type: GraphQLISODateTime,
+            description: `Most recent date contained in the field ${field.name}`,
+            fromField: field.name,
+            aggregationOperation: AGGREGATION_OPERATIONS.max,
+          },
+        },
+      ];
+    }
+
+    if (field.type === FieldMetadataType.NUMBER) {
```

Comment:
Only added NUMBER field so far, will handle CURRENCY in a different PR as it is a composite field and will require more work

Reply:
Yes every field should have an aggregation method (even uuid, etc) - cf initial issue for the fallback aggregation methods. Great if done in another PR!